### ISSUE TITLE
method `hash' called on hidden T_ARRAY object

### ISSUE THREAD

**Issue by stormsilver** [SamSaffron/memory_profiler#21]
Well, I've never seen something like this before. Don't even know if this would be considered a bug or not?

ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
memory_profiler revision 8a421fb8bb2c

```
method `hash' called on hidden T_ARRAY object (0x007fd6d0879d40 flags=0x20827)
/Users/ssilver/.rvm/gems/ruby-2.2.2/bundler/gems/memory_profiler-8a421fb8bb2c/lib/memory_profiler/results.rb:79:in `each'
/Users/ssilver/.rvm/gems/ruby-2.2.2/bundler/gems/memory_profiler-8a421fb8bb2c/lib/memory_profiler/results.rb:79:in `group_by'
/Users/ssilver/.rvm/gems/ruby-2.2.2/bundler/gems/memory_profiler-8a421fb8bb2c/lib/memory_profiler/results.rb:79:in `string_report'
/Users/ssilver/.rvm/gems/ruby-2.2.2/bundler/gems/memory_profiler-8a421fb8bb2c/lib/memory_profiler/results.rb:63:in `register_results'
/Users/ssilver/.rvm/gems/ruby-2.2.2/bundler/gems/memory_profiler-8a421fb8bb2c/lib/memory_profiler/reporter.rb:64:in `run'
/Users/ssilver/.rvm/gems/ruby-2.2.2/bundler/gems/memory_profiler-8a421fb8bb2c/lib/memory_profiler/reporter.rb:22:in `report'
/Users/ssilver/.rvm/gems/ruby-2.2.2/bundler/gems/memory_profiler-8a421fb8bb2c/lib/memory_profiler.rb:14:in `report'
```

**Comment by SamSaffron** [comment]
we got to guard against this.

**Comment by dgynn** [comment]
I believe this was caused by garbage collection and object ids being recycled during stat collection. This should be resolved with the 0.9.5 release. Please re-open if you see this again. Thanks!
